File: leetcode/leetcode75/724.java
class Solution {
    public int pivotIndex(int[] nums) {
        int left_sum = 0;
        int whole_sum = Arrays.stream(nums).sum();
        
        for (int i=0 ; i< nums.length ; i++) {
            if (left_sum * 2 + nums[i]== whole_sum) {
                return i;
            }
            left_sum += nums[i];
        }
        return -1;
    }
}

File: leetcode/leetcode75/724.py
class Solution(object):
    def pivotIndex(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        # whole_sum = sum(nums)
        # for i in range(len(nums)):
        #     left_sum = sum(num for num in nums[:i])
        #     if left_sum * 2 + nums[i] == whole_sum:
        #     # same as (if left_sum == right_sum)
        #         return i
        # return -1

        left_sum = 0
        right_sum = sum(nums)

        for idx, num in enumerate(nums):
            left_sum += num
            if left_sum == right_sum:
                return idx
            right_sum -= num
        return -1
